WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java

```diff
@@ -692,20 +689,67 @@ public String format(int value) {
         });
     }
 
-    private void showExportContentDialog() {
+    private void requestPurchasesForDeletionCheck() {
+        final Blog currentBlog = WordPress.getCurrentBlog();
+        WordPress.getRestClientUtils().getSitePurchases(currentBlog.getDotComBlogId(), new RestRequest.Listener() {
+            @Override
+            public void onResponse(JSONObject response) {
+                showPurchasesOrDeleteSiteDialog(response, currentBlog);
+            }
+        }, new RestRequest.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                ToastUtils.showToast(getActivity(), getString(R.string.purchases_request_error));
+                AppLog.e(AppLog.T.API, error.toString());
+            }
+        });
+    }
+
+    private void showPurchasesOrDeleteSiteDialog(JSONObject response, final Blog currentBlog) {
+        try {
```

Comment:
There is no chance it's null because `RestRequest` makes sure of it here: https://github.com/Automattic/android-wordpress-com-rest/blob/master/WordPressComRest/src/main/java/com/wordpress/rest/RestRequest.java#L112

If it can't parse the JSONObject (or it's null), it catches it and creates a new Response.Error object and returns that. So it's either good JSON or an Error.